server/src/main/java/org/elasticsearch/action/search/TransportSearchAction.java

```diff
@@ -1232,6 +1234,35 @@ private SearchPhase searchAsyncAction(
         }
     }
 
+    private static void resolveSearchEngine(ClusterState clusterState, IndexNameExpressionResolver resolver, SearchRequest searchRequest) {
+        if (searchRequest.searchEngineName() != null) {
+            return;
+        }
+
+        try {
+            List<String> searchEnginesName = resolver.searchEngineNames(
+                clusterState,
+                searchRequest.indicesOptions(),
+                searchRequest.indices()
+            );
+
+            if (searchEnginesName.size() > 0) {
+                if (searchEnginesName.size() > 1) {
+                    throw new IllegalArgumentException("Can't search several search engines together");
+                }
+
+                if (searchRequest.indices().length > 1) {
+                    // TODO: need a smarter version too handle wildcards. Enough for a POC.
+                    throw new IllegalArgumentException("Can't search several search engines together");
```

Comment:
I guess this should read:

```suggestion
                    throw new IllegalArgumentException("Can't search several search indices together");
```